src/pages/AdminPages/UnapprovedAccounts/SelectedUserModal.js

```diff
@@ -0,0 +1,145 @@
+                isUser : true,
+                isAffiliate : false,
+                isTrader : false
+            }, assignedTrader: traders[0].id}
+        }
+        if (role === 'affiliate') {
+            user = {...selectedUnapprovedUser, role:{
+                isAdmin : false,
+                isSuperAdmin : false,
+                isUser : true,
+                isAffiliate : true,
+                isTrader : false
+            }}
+        }
+        if (role === 'trader') {
+            user = {...selectedUnapprovedUser, role:{
+                isAdmin : false,
+                isSuperAdmin : false,
+                isUser : false,
+                isAffiliate : false,
+                isTrader : true
+            }}
+        }
+        dispatch(setSelectedUnApprovedUser(user))
+    }
+
+    const initialFormValues = {
+        firstName: selectedUnapprovedUser.firstName,
+        lastName: selectedUnapprovedUser.lastName,
+        email: selectedUnapprovedUser.email,
+        phone: selectedUnapprovedUser.phone,
+        affiliateCode: selectedUnapprovedUser.affiliateCode,
+        roi: "",
+    }
+
+    return (
+        <BaseModal 
+            title={`Approve ${selectedUnapprovedUser.email}'s account`}
+            show={showModal}
+            close={closeModal}
+            closeText="Cancel"
+        >
+            <Form
+                initialValues={initialFormValues}
+                onSubmit={approveUser}
+                validationSchema={validationSchema}
+             >
+                <FormField  name="firstName" ype="text" disabled />
+                <FormField name="lastName" type="text" disabled />
+                <FormField name="email" type="email" disabled />
+                <FormField name="phone" type="text" disabled />
+                <FormSelect name="role" index={0} 
+                    updateStuff={(role) => setRole(role)}
+                    options={role.isSuperAdmin ? 
+                        superAdminSettableRoles : adminSettableRoles} />
+                {selectedUnapprovedUser.role && (selectedUnapprovedUser.role.isUser || selectedUnapprovedUser.role.isAffiliate) && <>
+                    <FormField name="roi" placeholder="ROI" type="number" />
+                    <FormSelect name="assignedTrader" index={0} 
+                        options={traders.map(trader => 
+                            [`${trader.firstName} ${trader.lastName}`, trader.id])} />
+                </>}
+                {
+                    selectedUnapprovedUser.role && selectedUnapprovedUser.role.isAffiliate && 
+                    <FormField name="affiliateCode" placeholder="Affiliate Code" type="text" />
+                }
+                <div>
+                    <button onClick={() => {}}>Delete Account</button>
+                    <SubmitButton title="Approve" />
+                </div>
+            </Form>
+        </BaseModal>
+    )
+}
+
+export default SelectedUnapprovedUserModal
```

Comment:
I still have issues with the complexity of this file

Reply:
Most of what is in this file are functions, I've tried making the file smaller